This is a policy for the ARC milter

lquidfire opened the pull request:
This policy file particularly targets [OpenARC](https://github.com/flowerysong/OpenARC/). The OpenARC milter is both an open source library for adding Authenticated Received Chain (ARC) support to applications, and an example filter application using the milter protocol.

The policy files are based on the `dkim policy` in this repository, with adaptions where useful.

pebenito reviewed (changes requested):
Minor comments. Also see the lint errors from the GH actions.

lquidfire commented on policy/modules/services/arc.te:
```diff
@@ -0,0 +1,60 @@
+optional_policy(`
+	# set up unix socket
+	mta_sendmail_exec(openarc_milter_t)
+	#postfix_sendmail_exec(openarc_milter_t)  # Fron SELinux RefPolicy
```
Question:

In the `[dkim policy](https://github.com/SELinuxProject/refpolicy/blob/aa1d9dddcbc3403cded31ebfd42c72e56bef8b2e/policy/modules/services/dkim.te#L56)` file, it has the line `postfix_sendmail_exec(dkim_milter_t)`.

However, the `mta_sendmail_exec` line seems broader and more correct? Which one should be used?

pebenito replied:
Probably `mta_sendmail_exec()`, unless it specifically requires postfix. If it's generic, then removing it from the optional would also make sense.

lquidfire replied:
I had copied the wrong policy line. The optional policy was not for `mta_sendmail_exec`, but for `postfix_search_spool`. I must dig around, but am not sure that is needed for OpenARC.

`mta_sendmail_exec` is needed, however, for allowing OpenARC to send emails in case of an invalid signature. This was included in the [Fedora SELinux-policy](https://github.com/fedora-selinux/selinux-policy/blob/cdfbe797251b5ec9caa496aabb89e9ec2db46526/policy/modules/contrib/dkim.te#L37C1-L38C33) after a [bug report](https://bugzilla.redhat.com/show_bug.cgi?id=1716937).

lquidfire commented:
> Minor comments. Also see the lint errors from the GH actions.

I believe that I have addressed all issues. Thank you for reviewing it!

pebenito commented:
If you rename the module to openarc and squash to 1 commit, it should be good to go.

lquidfire commented:
> If you rename the module to openarc and squash to 1 commit, it should be good to go.

I changed the filenames, as well as the module name in `openarc.te`.

I squashed the commits, and force pushed them (I learned something new here). Thank you for your prompt assistance, this has been very helpful and a great experience for me!

lquidfire commented:
Postfix was not allowed to write to the openarc stream socket, so I added openarc optionally to `postfix.te` (following the example of `dkim` in `postfix.te`).

I then also  added `openarc` to `cloudinit.te` and `sysadm.te`, since I saw that `dkim` was also optionally added to those files, and it does make sense to add openarc to those.

pebenito commented:
Thanks!
